inc/pqueue.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+// Generic priority queue
```

Comment:
I know you mention it is min later but you should probably state if it is a min or max pqueue in the header description.